src/main/java/mate/academy/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+
+    @Override
+    @Transactional
+    public Order placeOrder(User user, ShippingAddressRequestDto shippingAddress) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findShoppingCartByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find shopping cart "
+                        + "by user id " + user.getId()));
+        double total = shoppingCart.getCartItems().stream()
+                .mapToDouble(cartItem -> (double) cartItem.getQuantity()
+                        * cartItem.getBook().getPrice().doubleValue())
+                .sum();
+        Order order = new Order();
+        order.setUser(user);
+        order.setShippingAddress(shippingAddress.getShippingAddress());
+        order.setOrderDate(LocalDateTime.now());
+        order.setStatus(Order.Status.NEW);
+        order.setTotal(BigDecimal.valueOf(total));
+        Set<OrderItem> orderItems = shoppingCart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    orderItem.setOrder(order);
+                    return orderItemRepository.save(orderItem);
+                })
+                .collect(Collectors.toSet());
+        shoppingCartRepository.delete(shoppingCart);
+        return orderRepository.save(order);
+    }
+
+    @Override
+    public List<OrderResponseDto> getAllOrders(User user,
+                                               Pageable pageable) {
+        Page<Order> allOrders = orderRepository.findAllByUserId(user.getId(), pageable);
+        return allOrders.stream()
+                .map(orderMapper::toResponseDto)
+                .toList();
+    }
+
+    @Override
+    public OrderResponseDto updateOrderStatus(Long orderId,
+                                              OrderStatusDto statusDto) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by id "
+                        + orderId));
+        order.setStatus(statusDto.getStatus());
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public List<OrderItemResponseDto> getAllOrderItems(User user, Long orderId, Pageable pageable) {
+        Order order = orderRepository.findByUserIdAndId(user.getId(), orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't find order by order id "
+                        + orderId + " and user id " + user.getId()));
+        return (List<OrderItemResponseDto>) orderItemMapper
+                .toResponseDto((OrderItem) order.getOrderItems());
```

Comment:
Incorrect approach. You need to implement mapping collections to your orderItemMapper